app/helpers/application_helper.rb

```diff
@@ -1,2 +1,24 @@
 module ApplicationHelper
+  def bootstrap_class_for(flash_class)
+    { success: "alert-success",
+      error: "alert-danger",
+      alert: "alert-warning",
+      notice: "alert-info" }[flash_class.to_sym] || flash_class.to_s
+  end
+
+  def flash_messages(opts = {})
+    flash.each do |msg_type, message|
+      concat(content_tag(
+        :div, message,
+        class: "alert #{bootstrap_class_for(msg_type)} alert-dismissible",
+          role: "alert") do
+        concat(content_tag(:button, class: "close", data: { dismiss: "alert" }) do
+          concat content_tag(:span, "&times;".html_safe, "aria-hidden" => true)
+          concat content_tag(:span, "Close", class: "sr-only")
+        end)
+        concat message
+      end)
```

Comment:
`end` at 20, 6 is not aligned with `content_tag(` at 11, 13 or `role: "alert") do` at 14, 10